### ISSUE TITLE
running switchdiscvoer on a mellanox driver seems to take a long time or locksup

### ISSUE THREAD

**Issue by ralphbellofatto** [xcat2/xcat-core#3989]
```
[root@c699mgt00 ~]# switchdiscover --range 10.5.254.1-250 -s snmp -w --setup
Discovering switch using snmpwalk for 10.5.254.1-250 .....
ip              name                    vendor                                                  mac  
------------    ------------            ------------                                            ------------
10.5.254.9      switch-94d166           Mellanox CS7520,MLNX-OS,SWv3.6.3004                     24:8a:07:94:d1:66
10.5.254.8      switch-a40734           Mellanox CS7520,MLNX-OS,SWv3.6.3004                     24:8a:07:a4:07:34
switch discovered and matched: switch-94d166 to c699ibdir3a
switch discovered and matched: switch-a40734 to c699ibdir1a
Writing the data into xCAT DB....
call to config Mellanox switches c699ibdir3a,c699ibdir1a

```

We started switch discover about 30 minutes ago, and it is still running.

How long should this take to complete?

**Comment by ralphbellofatto** [comment]
@cxhong  adding interested party.

**Comment by ralphbellofatto** [comment]
switch discover never finished...  when we run it on only one node at a time it also never finishes..

**Comment by cxhong** [comment]
yeah, it caused by xdsh command never return back after set to new ip address
I opened issue #3992 for this.

**Comment by ralphbellofatto** [comment]
so we need to figure out two things, why we are not getting a time out and 2, why do these switches not return...

**Comment by cxhong** [comment]
I tested same steps on our test system.
````
# switchdiscover --range 192.168.5.49 --setup
[fs1]: Discovering switches using nmap for 192.168.5.49. It may take long time...
[fs1]: ip               name                    vendor                                                  mac 
[fs1]: ------------     ------------            ------------                                            ------------
[fs1]: 192.168.5.49     switch-192-168-5-49     Mellanox IB switch                                      7C:FE:90:2C:B8:F8
[fs1]: switch discovered and matched: switch-192-168-5-49 to c910f05ibsw1mgt4
[fs1]: call to config Mellanox switches c910f05ibsw1mgt4

[fs1]: output = c910f05ibsw1mgt4: Changing IP address to static IP 192.168.5.50
c910f05ibsw1mgt4: Changing host name to c910f05ibsw1mgt4
c910f05ibsw1mgt4: snmpv3 configured
`````
the switch is configured and no issue with xdsh timeout.

**Comment by ralphbellofatto** [comment]
was this the same exact switch AND was it in the factory default configuration?

**Comment by cxhong** [comment]
the Mellanox command "no interface mgmt0 dhcp"  caused switch hung.  Jame will upgrade this switch to newest firmware level, then we will try again.

**Comment by cxhong** [comment]
firmware upgrade didn't resolve this issue.  
when we issued the command one by one on the switch,  we noticed "no interface mgmt0 dhcp" will remove ip address from mgmt0,  that cause switch hung and unreachable.   only way set ip address back is  from serial connection.
Actually, change ip address from xCAT to send serial commands to switch works better,  ip address changed, can be reached from other terminal.    xCAT need to figure out way to return back from xdsh command.
